pkg/storages/azure/folder.go

```diff
@@ -58,40 +63,109 @@ func NewCredentialError(settingName string) storage.Error {
+	containerUrlString := fmt.Sprintf("https://%s.blob.%s/%s%s", accountName, storageEndpointSuffix, containerName, accountToken)
+	_, err := url.Parse(containerUrlString)
+	if err != nil {
+		return nil, NewFolderError(err, "Unable to parse service URL with SAS token")
+	}
+
+	containerClient, err := azblob.NewContainerClientWithNoCredential(containerUrlString, &azblob.ClientOptions{
+		Retry: policy.RetryOptions{TryTimeout: timeout},
+	})
+	return containerClient, err
+}
+
+func getContainerClientWithAccessKey(
+	accountName string,
+	storageEndpointSuffix string,
+	containerName string,
+	timeout time.Duration,
+	credential *azblob.SharedKeyCredential) (*azblob.ContainerClient, error) {
+	containerUrlString := fmt.Sprintf("https://%s.blob.%s/%s", accountName, storageEndpointSuffix, containerName)
+	_, err := url.Parse(containerUrlString)
+	if err != nil {
+		return nil, NewFolderError(err, "Unable to parse service URL")
+	}
+
+	containerClient, err := azblob.NewContainerClientWithSharedKey(containerUrlString, credential, &azblob.ClientOptions{
+		Retry: policy.RetryOptions{TryTimeout: timeout},
+	})
+	return containerClient, err
+}
+
+func getContainerClient(
+	accountName string,
+	storageEndpointSuffix string,
+	containerName string,
+	timeout time.Duration) (*azblob.ContainerClient, error) {
+	defaultCredential, err := azidentity.NewDefaultAzureCredential(nil)
+	if err != nil {
+		return nil, NewFolderError(err, "Unable to construct default Azure credential chain")
+	}
+
+	containerUrlString := fmt.Sprintf("https://%s.blob.%s/%s", accountName, storageEndpointSuffix, containerName)
+	_, err = url.Parse(containerUrlString)
+	if err != nil {
+		return nil, NewFolderError(err, "Unable to parse service URL")
 	}
-	if accountKey, ok = settings[AccessKeySetting]; !ok {
-		if accountToken, usingToken = settings[SasTokenSetting]; !usingToken {
-			return nil, NewCredentialError(AccessKeySetting)
-		}
 
+	containerClient, err := azblob.NewContainerClient(containerUrlString, defaultCredential, &azblob.ClientOptions{
+		Retry: policy.RetryOptions{TryTimeout: timeout},
+	})
+	return containerClient, err
+}
+
+func configureAuthType(settings map[string]string) (AzureAuthType, string, string) {
+	var ok bool
+	var accountToken, accessKey string
+	var authType AzureAuthType
+
+	if accessKey, ok = settings[AccessKeySetting]; ok {
+		authType = AzureAccessKeyAuth
+	} else if accountToken, ok = settings[SasTokenSetting]; ok {
+		authType = AzureSASTokenAuth
 		// Tokens may or may not begin with ?, normalize these cases
 		if !strings.HasPrefix(accountToken, "?") {
 			accountToken = "?" + accountToken
 		}
 	}
 
+	return authType, accountToken, accessKey
+}
+
+func ConfigureFolder(prefix string, settings map[string]string) (storage.Folder, error) {
+	var accountName, accountKey, accountToken, storageEndpointSuffix string
+	var ok bool
+	var authType AzureAuthType
+	if accountName, ok = settings[AccountSetting]; !ok {
```

Comment:
I suggest using the ':=' instead to avoid the unnecessary declaration above.

Reply:
The accountName and ok declarations are actually necessary, but I'm not familiar enough with go to articulate why. The build fails with `accountName declared but not used`, `undefined: accountName`, and `undefined: ok` errors when I change it.